Improves keyboard navigation accessibility

HaSistrunk opened the pull request:
- Changes strategy for main site navigation (using this [ARIA Authoring Practices example pattern](https://w3c.github.io/aria-practices/examples/disclosure/disclosure-navigation.html)) for better keyboard navigation options (using arrow keys, tab, and esc), and based on research on most commonly used a11y navigation patterns.
- Improves focus styles for keyboard navigation in main nav and side nav.
- Fixes #72 to add keyboard focus when not using a mobile device.
- Removes `inclusive-menu-button.js` and adjusts table of contents to use links instead of buttons

HaSistrunk committed:
improve top nav accessibility

HaSistrunk committed:
remove button outline when menu expands

HaSistrunk committed:
enable keyboard focus in side nav, fixes #72

HaSistrunk committed:
add sticky position support for Safari

helrond commented on _includes/scripts.html:
```diff
@@ -1,5 +1,6 @@
 <script src="https://code.jquery.com/jquery-3.3.1.min.js" crossorigin="anonymous"></script>
 
+<script src="{{ "/assets/js/disclosure-menu.min.js" | relative_url }}"></script>
 <script src="{{ "/assets/js/inclusive-menu-button.min.js" | relative_url }}"></script>
```
Does the script you added replace the previous JS (inclusive-menu-button.min.js)? If so we can get rid of this script import (and the script that's imported)

HaSistrunk replied:
The script I added replaces the inclusive-button.min.js for the top nav, but the side nav is still using it (and I think it is probably more appropriate for the side nav).

helrond replied:
Hmmm, I'm not a huge fan of scripts with overlapping functionality which serve different parts of the page. The more stuff we load on a page the slower it is, and the more possibilities there are for conflicts. Is there a way we can use one script for both purposes?

HaSistrunk replied:
I think so, yes, but I was struggling some to understand how the side nav was working with the various scripts. I can revisit this.

helrond replied:
Great, I want to make sure we're not unnecessarily creating technical debt.

helrond reviewed (changes requested):
Added one detailed comment but two other things:
1. when I've opened a menu using keyboard navigation, I am unable to close it and navigate to another top menu item. The spacebar scrolls, and the arrow keys seem boxed into the expanded menu items.
2. There's a weird styling thing happen with the open menus. I've tried flushing the cache, but to no avail.
<img width="483" alt="Screen Shot 2020-05-05 at 9 55 38 AM" src="https://user-images.githubusercontent.com/607621/81081099-1b83c000-8eb7-11ea-8a94-6970cc8d2336.png">

HaSistrunk committed:
fix styling in dropdown menu

HaSistrunk commented:
> when I've opened a menu using keyboard navigation, I am unable to close it and navigate to another top menu item. The spacebar scrolls, and the arrow keys seem boxed into the expanded menu items.

Yes, so I was reading through [the PR discussion for this pattern](https://github.com/w3c/aria-practices/pull/1036) leading to its addition as an example to the ARIA Authoring Practices. Originally there was no support for arrow keys, since the default for moving keyboard focus for screen reader use is `Tab / Shift + Tab`. However, the optional use of arrow keys to traverse open menu items was added for keyboard users who are not running a screen reader. `Esc` closes the menu and returns focus to the top-level button.

I can't find much info on user testing around these non-screenreader keyboard use patterns, but from my research on accessible link navigation with disclosure widgets, I like this pattern and it seems to be more widely used than using `role="menu"` for web page navigation (backed up by usa.gov, which uses a version of this without arrow key support).

HaSistrunk commented:
I _think_ I fixed the weird styling issue with the open menu with my last commit. Thanks for catching that!

helrond commented:
This looks good to me! If we can get that issue with the scripts resolved I think this is good to go.

One thing I want to pay attention to on deployment is whether the new CSS styles get loaded in production without clearing the browser cache. If they don't we'll want to look at that and see if we can force that to happen somehow.

helrond committed:
Merge pull request #99 from RockefellerArchiveCenter/development

Add AAPI statement

helrond committed:
Merge branch 'development' into base

HaSistrunk committed:
update toc to work w/out incl menu button script

HaSistrunk committed:
implement aria attributes for disclosure buttons

HaSistrunk committed:
remove toc link underline

HaSistrunk committed:
use more than color for keyboard focus styles

HaSistrunk commented:
> This looks good to me! If we can get that issue with the scripts resolved I think this is good to go.
> 
> One thing I want to pay attention to on deployment is whether the new CSS styles get loaded in production without clearing the browser cache. If they don't we'll want to look at that and see if we can force that to happen somehow.

OK! I've removed the `inclusive-menu-button.js` script and updated the existing scripts to use links instead of buttons on the side nav, which simplifies that some.

HaSistrunk committed:
Merge branch 'base' into issue-72

helrond committed:
formatting

helrond committed:
ignore jekyll cache

helrond committed:
add alt text for logo
